pq-crypto/s2n_pq_asm.h

```diff
@@ -15,4 +15,8 @@
 
 #pragma once
 
-#define S2N_ANY_NONPORTABLE_OPTIMIZATIONS_ENABLED   __AVX__ || __AVX2__ || __BMI2__
+#ifdef S2N_ALLOW_NONPORTABLE_OPTIMIZATIONS
+    #define S2N_ANY_NONPORTABLE_OPTIMIZATIONS_ENABLED   0
```

Comment:
Naming doesn't really make sense - setting `S2N_ALLOW_NONPORTABLE_OPTIMIZATIONS` makes `S2N_ANY_NONPORTABLE_OPTIMIZATIONS_ENABLED` set to 0, which is likely not true. How about renaming it to `S2N_PROHIBITED_NONPORTABLE_OPTIMIZATIONS_DETECTED`